traits_futures/testing/gui_test_assistant.py

```diff
@@ -73,3 +86,25 @@ def run_until(self, object, trait, condition, timeout=SAFETY_TIMEOUT):
             true or not at that point.
         """
         self._event_loop_helper.run_until(object, trait, condition, timeout)
+
+    def exercise_event_loop(self):
+        """
+        Exercise the event loop.
+
+        Places a new task on the event loop and runs the event loop
+        until that task is complete. The goal is to flush out any other
+        tasks that might already be in event loop tasks queue.
+
+        Note that there's no guarantee that this will execute other pending
+        event loop tasks. So this method is useful for tests of the form
+        "check that nothing bad happens as a result of other pending event
+        loop tasks", but it's not safe to use it for tests that *require*
+        pending event loop tasks to be processed.
+        """
+        sentinel = _HasBool()
+        self._event_loop_helper.setattr_soon(sentinel, "flag", True)
+        self.run_until(
+            sentinel,
+            "flag",
+            lambda sentinel: sentinel.flag,
+        )
```

Comment:
i suspect black will complain about this otherwise

```suggestion
        self.run_until(sentinel, "flag", lambda sentinel: sentinel.flag)
```

Reply:
Black's actually fine with this (at least, the current version of black from PyPI is; the version in EDM may not be). But yes, this would be better on one line.